apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -909,6 +910,14 @@ handle_request_('GetPinningTx', _Params, _Context) ->
            {404, [], #{reason => <<"No pin data available">>}}
     end;
 
+handle_request_('GetHyperchainContractPubkeys', _Params, _Context) ->
+    % TODO handle not in HC at all or consensus not initialized?
+    lager:debug("HyperchainsGetContract"),
+    {200, [], #{<<"staking">> => aeser_api_encoder:encode(contract_pubkey, aec_consensus_hc:get_contract_pubkey(staking)),
```

Comment:
Here we spill over the logic on how many contracts there are and what they are named from consensus to the API. The names  `election`, `rewards` are macros in the consensus code, but now hard-coded in the API.

Not sure about this, but my feeling is that it would be cleaner to ask the consensus module for all contracts and have that module return this map. In that way, the logic of how many and what contracts we have can be adapted rather easily.... but the price is that we cannot say in the API specification what we expect back other than an object with keys and contract addresses.

Reply:
Sure, can do that no prob. Not sure how the endpoint will feel about getting a map with unspeced properties though. What's the yaml construct for that?